src/gsi/hybrid_ensemble_isotropic.F90

```diff
@@ -3624,34 +3634,29 @@ subroutine bkerror_a_en(grady)
   call sqrt_beta_e_mult(grady)
 
 ! Apply variances, as well as vertical & horizontal parts of background error
-!   !$omp parallel do schedule(dynamic,1) private(ii)
-  do ii=1,nsubwin
-     if (naensgrp==1) then
+  if (naensgrp==1) then
+     do ii=1,nsubwin
         call bkgcov_a_en_new_factorization(1,grady%aens(ii,1,1:n_ens))
-     else
-        allocate(z(naensgrp,nval_lenz_en))
+     end do
+  else
+     allocate(z(nval_lenz_en,naensgrp))
+     allocate(z2(nval_lenz_en))
+     do ii=1,nsubwin
         do ig=1,naensgrp
-           call ckgcov_a_en_new_factorization_ad(ig,z(ig,:),grady%aens(ii,ig,1:n_ens))
+           call ckgcov_a_en_new_factorization_ad(ig,z(1,ig),grady%aens(ii,ig,1:n_ens))
```

Comment:
@jderber-NOAA Is using z(1,ig) here instead of z(:,ig) intentional?

Reply:
Good question!!!

Yes, it actually was.  What I will say here is from my knowledge from a few (maybe more) years ago.  It may have changed since then.

Both should result in the same answer.  However when the ":" is used the compiler will have it create a copy of the data before passing it to the subroutine.  When "1" is used the compiler will just pass the starting address to the subroutine.  The second is faster if the data is already sequential.  If the ":" is, for example, in the second dimension, it may be faster because it will bring the data into a new array and will make the data sequential in memory in the subroutine.

Reply:
Thanks @jderber-NOAA!